src/main/scala/firrtl/Visitor.scala

```diff
@@ -58,6 +58,56 @@ class Visitor(infoMode: InfoMode) extends AbstractParseTreeVisitor[FirrtlNode] w
   private def string2Int(s: String): Int = string2BigInt(s).toInt
 
   private def visitInfo(ctx: Option[InfoContext], parentCtx: ParserRuleContext): Info = {
+    // Convert a compressed FileInfo string into either into a singular FileInfo or a MultiInfo
+    // consisting of several FileInfos
+    def parseCompressedInfo(escaped: String): Info = {
+      var out: Seq[FileInfo] = Seq()
+
+      // Regular expression to match and capture the general File.format line:col pattern.
+      // Also matches the remaining part of the string which doesn't match the expression;
+      // which will be passed directly into the output as a FileInfo.
+      val splitCompressedInfo = """([^\s:]+)((?: \d+:(?:\d+|\{\d+(?:,\d+)+\}))+)|(?:[^\s].*)""".r
+
+      // Regular expression to capture the line number and column numbers in the compressed file info pattern.
+      val splitLineDescriptors = """(\d+):((?:\d+|\{\d+(?:,\d+)+\}))""".r
+
+      // Regular expression to match against individual column numbers in each line:col or line:{col1,col2}
+      // descriptor.
+      val splitColDescriptors = """\d+""".r
+
+      val matches = splitCompressedInfo.findAllIn(escaped)
+
+      // Grab each File.format line:col token from the input string
+      splitCompressedInfo.findAllIn(escaped).matchData.foreach { info =>
+        {
+          Option(info.group(1)) match {
+            // If there were no subgroups, the regex matched against a non-conforming source locator
+            // pattern, so do not process it
+            case None => out = out :+ ir.FileInfo.fromEscaped(info.toString)
+            case Some(file) => {
```

Comment:
```suggestion
            case Some(file) =>
```
Again, unnecessary curly braces.